pkg/util/patroni/patroni.go

```diff
@@ -43,8 +44,23 @@ func New(logger *logrus.Entry) *Patroni {
 	}
 }
 
-func apiURL(masterPod *v1.Pod) string {
-	return fmt.Sprintf("http://%s:%d", masterPod.Status.PodIP, apiPort)
+func apiURL(masterPod *v1.Pod) (string, error) {
+	ip := net.ParseIP(masterPod.Status.PodIP)
+	if ip == nil {
+		return "", fmt.Errorf("%s is not a valid IP", masterPod.Status.PodIP)
+	}
+	var ipString string
+	// If IPv6, add braces
+	if ip.To4() == nil {
+		if ip.To16() == nil {
+			// Shouldn't ever get here, but library states it's possible.
+			return "", fmt.Errorf("%s is not a valid IPv4/IPv6 address", ip.String())
+		}
+		ipString = fmt.Sprintf("[%s]", ip.String())
```

Comment:
Probably we can even use [JoinHostPort](https://golang.org/pkg/net/#JoinHostPort) here, as it will take care of brackets.

Reply:
Good catch. Do you think it's worth keeping the IP validation since `JoinHostPort` handles both?

Reply:
Good point. Unfortunately I can't find any direct confirmation in K8S docs or code, that `PodIP` always contain correct `IP` string. Looks like in some cases `IP` could be not assigned and `PodIP` would contain an empty string, but I'm not sure if we can face such cases. Maybe you have some references, that proves correctness of IP address? Otherwise probably it makes sense to keep some validation just in case.